velox/exec/Task.cpp

```diff
@@ -917,13 +923,17 @@ void Task::resume(std::shared_ptr<Task> self) {
             continue;
           }
           VELOX_CHECK(!driver->isOnThread() && !driver->isTerminated());
-          if (!driver->state().hasBlockingFuture) {
+          if (!driver->state().hasBlockingFuture &&
+              driver->task()->queryCtx()->isExecutorSupplied()) {
```

Comment:
You can check Task::mode_ here.

Reply:
Planned to do this with some other cleanups in later PR. Does that work for you?